internal/resource/config/localidentity/local_identity_identity_profiles_resource.go

```diff
@@ -0,0 +1,917 @@
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.ProfileConfig, false)), addRequest.ProfileConfig)
+		fmt.Println(err)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.FieldConfig) {
+		addRequest.FieldConfig = client.NewFieldConfig()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.FieldConfig, false)), addRequest.FieldConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.EmailVerificationConfig) {
+		addRequest.EmailVerificationConfig = client.NewEmailVerificationConfigWithDefaults()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.EmailVerificationConfig, true)), addRequest.EmailVerificationConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsNonEmptyObj(plan.DataStoreConfig) {
+		addRequest.DataStoreConfig = client.NewLdapDataStoreConfigWithDefaults()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.DataStoreConfig, false)), addRequest.DataStoreConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.ProfileEnabled) {
+		addRequest.ProfileEnabled = plan.ProfileEnabled.ValueBoolPointer()
+	}
+	return nil
+}
+
+// Metadata returns the resource type name.
+func (r *localIdentityIdentityProfilesResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_local_identity_identity_profiles"
+}
+
+func (r *localIdentityIdentityProfilesResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+func (r *localIdentityIdentityProfilesResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+
+	var model localIdentityIdentityProfilesResourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+	// Validates Email Verification type for Email Configuration
+	if internaltypes.IsDefined(model.EmailVerificationConfig) {
+		emailVerificationConfig := model.EmailVerificationConfig.Attributes()
+		emailVerificationType := model.EmailVerificationConfig.Attributes()["email_verification_type"].(basetypes.StringValue).ValueString()
+		switch emailVerificationType {
+		case "OTP":
+			if internaltypes.IsDefined(emailVerificationConfig["otl_time_to_live"].(basetypes.Int64Value)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("otl_time_to_live attribute is not allowed when email_verification_type is OTP. Required attributes are otp_length, otp_retry_attempts and otp_time_to_live."))
+			}
+			if internaltypes.IsDefined(emailVerificationConfig["otp_length"].(basetypes.Int64Value)) && (emailVerificationConfig["otp_length"].(basetypes.Int64Value).ValueInt64() < 5) {
```

Comment:
If someone was to pass in a non-int value for otp_length in HCL for example, would this cause a panic? I don't think it would but it would be good to check. It feels a little risky to have all these casts everywhere.